### PR TITLE
Add http legos

### PR DESCRIPTION
**Pull request by GH0st3rs**

### CODE REVIEW

File: katnip/legos/http.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            Static(': '),
+            Container(name='%s value' % key, fields=[value]),
+            Static('\r\n')
+        ]
+        if end:
+            fields += [Static('\r\n')]
+        super(HttpHeaderField, self).__init__(name=key, fields=fields, fuzzable=fuzzable)
+
+
+class textField(HttpHeaderField):
```
Comment (BinyaminSharet) [inline katnip/legos/http.py]: Class name starts with a capital letter - TextField

File: katnip/legos/http.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    def __init__(self, key, value, end=False, fuzzable=True):
+        value_field = String(name="value", value=value)
+        super(textField, self).__init__(key, value_field, end, fuzzable=fuzzable)
+
+
+class intField(HttpHeaderField):
```
Comment (BinyaminSharet) [inline katnip/legos/http.py]: Class name starts with a capital letter - IntField

File: katnip/legos/url.py [hunk 3]
```diff
@@ -243,15 +248,22 @@ def __init__(self, search='', fuzz_delims=False, fuzzable=True, name=None):
         '''
         fields = [
             Delimiter(name='search main delim', value='?', fuzzable=fuzz_delims),
-            String(name='search data', value=search),
         ]
+        for part in search.split('&'):
+            part = part.split('=')
+            fields.append(Container(name='param_%s' % part[0], fields=[
+                String(name=part[0], value=part[0], fuzzable=fuzz_param),
```
Comment (BinyaminSharet) [inline katnip/legos/url.py]: change name to something like 'search_%d_key' % (i)

File: katnip/legos/url.py [hunk 4]
```diff
@@ -243,15 +248,22 @@ def __init__(self, search='', fuzz_delims=False, fuzzable=True, name=None):
...
+                Delimiter(value='=', fuzzable=fuzz_delims),
+                String(name=part[1], value=part[1], fuzzable=fuzz_value)
```
Comment (BinyaminSharet) [inline katnip/legos/url.py]: change name to something like 'search_%d_value' % (i)

File: katnip/legos/url.py [hunk 5]
```diff
@@ -265,6 +277,8 @@ def __init__(self, path=None, path_delim='/', fuzz_delims=True, fuzzable=True, n
         if path is not None:
             fields.append(Delimiter(name='main delim', value='/', fuzzable=fuzz_delims))
             path_parts = path.split(path_delim)
+            if not path_parts[0]:
```
Comment (BinyaminSharet) [inline katnip/legos/url.py:280]: better to do path_delim.strip() on the previous line

Comment (GH0st3rs) [reply katnip/legos/url.py:280]: If `path="/index"` then
`path_parts = ["", "index"]`

If `path="index"` then
`path_parts = ["index"]`

This code: `if not path_parts[0]:` need for 1 variant

**Pull request merged**